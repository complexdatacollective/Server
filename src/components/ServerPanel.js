import React, { Component } from 'react';
import { PanelItem } from '../components';
import Server from '../utils/Server';

const defaultServerOverview = {
  ip: 'x.x.x.x',
  clients: 0,
  uptime: 0,
  publicKey: '',
};

class ServerPanel extends Component {
  constructor() {
    super();

    this.state = {
      version: '0.0.0',
      notes: '',
    };

    this.server = new Server();

    this.server.on('SERVER_STATUS', (data) => {
      this.setState({ serverOverview: data });
    });

    this.server.requestServerStatus();
  }

  render() {
    const { serverOverview } = this.state;

    const overview = { ...defaultServerOverview, ...serverOverview };
    return (
      <div className="server-panel">
        <PanelItem label="IP" value={JSON.stringify(overview.ip)} />
        <PanelItem label="Clients" value={overview.clients} />
        <PanelItem label="Uptime" value={overview.uptime} />
        <PanelItem label="Public Key" value={overview.publicKey} />
      </div>
    );
  }
}

export default ServerPanel;
