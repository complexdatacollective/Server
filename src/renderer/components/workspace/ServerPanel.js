import React from 'react';
import PropTypes from 'prop-types';
import DeviceStatus from '../../containers/DeviceStatus';
import NetworkStatus from '../../containers/NetworkStatus';

const ServerPanel = ({ className }) => (
  <div className={`server-panel ${className}`}>
    <div className="server-panel__wrapper">
      <NetworkStatus />
      <DeviceStatus />
    </div>
  </div>
);

ServerPanel.defaultProps = {
  className: '',
};

ServerPanel.propTypes = {
  className: PropTypes.string,
};

export {
  ServerPanel as UnwrappedServerPanel,
};

export default ServerPanel;
