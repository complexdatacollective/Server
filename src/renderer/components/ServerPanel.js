import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { PanelItem } from '../components';
import withApiClient from './withApiClient';

class ServerPanel extends Component {
  constructor() {
    super();
    this.state = {
      version: '0.0.0',
    };
  }

  componentWillMount() {
    this.getServerHealth();
  }

  componentDidUpdate() {
    this.getServerHealth();
  }

  getServerHealth() {
    const { apiClient } = this.props;
    if (!apiClient || this.state.serverOverview) {
      return;
    }
    apiClient.get('/health')
      .then((resp) => {
        const {
          deviceApiPort,
          hostname,
          ip,
          isAdvertising,
          mdnsIsSupported,
          uptime,
        } = resp.serverStatus;
        let mdnsStatus = isAdvertising ? 'Active' : 'Pending';
        if (!mdnsIsSupported) {
          mdnsStatus = 'Unsupported';
        }
        this.setState({
          serverOverview: {
            hostname,
            ip: ip && ip.address,
            deviceApiPort,
            mdnsStatus,
            uptime,
          },
        });
      })
      .catch(() => {
        this.setState({
          serverOverview: {},
        });
      });
  }

  render() {
    const { serverOverview } = this.state;
    const { className } = this.props;
    const overview = { ...serverOverview };
    const uptimeDisplay = overview.uptime && `${parseInt(overview.uptime / 1000 / 60, 10)}m`;
    return (
      <div className={`server-panel ${className}`}>
        <PanelItem label="Local Server IP" value={overview.ip || 'Offline'} />
        <PanelItem label="Server Port" value={overview.deviceApiPort || '-'} />
        <PanelItem label="Uptime" value={uptimeDisplay || '-'} />
        <PanelItem label="Server Hostname" value={overview.hostname || '-'} />
        <PanelItem label="Service Advertising" value={overview.mdnsStatus || '-'} />
      </div>
    );
  }
}

ServerPanel.defaultProps = {
  apiClient: null,
  className: '',
};

ServerPanel.propTypes = {
  apiClient: PropTypes.object,
  className: PropTypes.string,
};

export default withApiClient(ServerPanel);

export {
  ServerPanel as UnwrappedServerPanel,
};
