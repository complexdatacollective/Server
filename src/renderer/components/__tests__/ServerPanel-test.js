/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { UnwrappedServerPanel as ServerPanel } from '../ServerPanel';

jest.mock('../../utils/adminApiClient');

const mockServerOverview = {
  ip: 'x.x.x.x',
  clients: 0,
  uptime: 0,
  publicKey: '',
};
// setting Date.now so that it is consistent for snapshot
Date.now = jest.fn(() => 1482363367071);

describe('<ServerPanel />', () => {
  it('should render', () => {
    const serverPanel = shallow((
      <ServerPanel serverOverview={mockServerOverview} />
    ));

    expect(serverPanel).toMatchSnapshot();
  });
});
