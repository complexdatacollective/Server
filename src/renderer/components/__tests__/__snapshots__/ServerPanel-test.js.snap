// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerPanel /> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServerPanel
    className=""
    serverOverview={
      Object {
        "clients": 0,
        "ip": "x.x.x.x",
        "publicKey": "",
        "uptime": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PanelItem
          label="Server Public IP"
          value="\\"x.x.x.x\\""
        />,
        <PanelItem
          label="Uptime"
          value={0}
        />,
        <PanelItem
          label="Server Public Key"
          value=""
        />,
      ],
      "className": "server-panel ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Server Public IP",
          "value": "\\"x.x.x.x\\"",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Uptime",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Server Public Key",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PanelItem
            label="Server Public IP"
            value="\\"x.x.x.x\\""
          />,
          <PanelItem
            label="Uptime"
            value={0}
          />,
          <PanelItem
            label="Server Public Key"
            value=""
          />,
        ],
        "className": "server-panel ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Server Public IP",
            "value": "\\"x.x.x.x\\"",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Uptime",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Server Public Key",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
